odl/tomo/backends/astra_cpu.py

```diff
@@ -107,12 +103,20 @@ def astra_cpu_forward_projector(vol_data, geometry, proj_space, out=None):
     proj_id = astra_projector(vol_interp, vol_geom, proj_geom, ndim,
                               impl='cpu')
 
-    # Create algorithm
-    algo_id = astra_algorithm('forward', ndim, vol_id, sino_id, proj_id,
-                              impl='cpu')
+    # Create ASTRA data structures
+    vol_id = astra_data(vol_geom, datatype='volume', data=vol_data,
+                        allow_copy=True)
 
-    # Run algorithm
-    astra.algorithm.run(algo_id)
+    with writable_array(out, dtype='float32', order='C') as arr:
```

Comment:
Was slightly skeptical to this "functional" approach, but I clearly see the advantage with array-like input.